### PR TITLE
Implement tox

### PR DESCRIPTION
**Pull request by KasukabeDefenceForce**
### :pencil: Description

**Type:** :rocket: `feature`
The PR aims at implementing tox for automated testing across various Python versions and environments. 

### :pushpin: Resources

Examples, notebooks, and links to useful references.

### :vertical_traffic_light: Testing

How did you test these changes?

- [ ] Testing pipeline
- [ ] Other method (describe)
- [ ] My changes can't be tested (explain why)

### :ballot_box_with_check: Checklist

- [ ] I requested two reviewers for this pull request
- [ ] I updated the documentation according to my changes
- [ ] I built the documentation by applying the `build_docs` label

> **Note:** If you are not allowed to perform any of these actions, ping (@) a contributor.

### CODE REVIEW

**Comment by KasukabeDefenceForce** [comment]
For running tests on all the mentioned environments use the command:
tox
For running tests on specific environment use the command:
tox -e {env} -test

File: tardis/visualization/tools/tests/test_sdec_plot.py [hunk 1]
```diff
@@ -527,15 +541,15 @@ def test_generate_plot_mpl(
...
-        species_list,
+        request: pytest.FixtureRequest,
+        plotter: SDECPlotter,
+        packets_mode: Literal["virtual"] | Literal["real"],
+        packet_wvl_range: Any | None,
+        distance: Any | None,
+        show_modeled_spectrum: bool,
+        observed_spectrum: tuple,
+        nelements: None | Literal[1],
+        species_list: list[str] | None,
```
Comment (atharva-2001) [inline tardis/visualization/tools/tests/test_sdec_plot.py:552]: I don't think we use type annotations anymore

**Comment by andrewfullard** [comment]
You said in Slack that the main benefit is testing against multiple Python versions. With our static environment, I am not sure that applies to TARDIS. What other benefits does tox provide?

**Comment by KasukabeDefenceForce** [comment]
> You said in Slack that the main benefit is testing against multiple Python versions. With our static environment, I am not sure that applies to TARDIS. What other benefits does tox provide?

Some other benefits:
1. We can simplify the test command. For example, instead of running `pytest tardis --tardis-refdata=/Users/abhinav/workspace/code/tardis-main/tardis-refdata --tardis-regression-data=/Users/abhinav/workspace/code/tardis-main/tardis-regression-data` we can just run `tox`.
2. Tox creates separate virtual environments for each test, preventing dependency conflicts and ensuring consistent test conditions. For example, if  a user has installed some extra dependencies locally, it wont affect the test environment.

Resources:
https://blog.ionelmc.ro/2015/04/14/tox-tricks-and-patterns/
https://codilime.com/blog/python-tox/
https://christophergs.com/python/2020/04/12/python-tox-why-use-it-and-tutorial/

**Comment by wkerzendorf** [comment]
is this done? @atharva-2001 told me that we don't want to use it.

**Comment by KasukabeDefenceForce** [comment]
> Can tox implement single-command testing for all of the not (continuum or rpacket_tracking), continuum, and rpacket_tracking test cases?

Yes we can.

**Comment by KasukabeDefenceForce** [comment]
> is this done? @atharva-2001 told me that we don't want to use it.

Yep, its not worth it in our case.

**Pull request closed without merging**